File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexes_hashed = {}
        for i, x in enumerate(nums):
            required = target-x
            if required in indexes_hashed:
                return [indexes_hashed[required], i]
            indexes_hashed[x] = i
        return []
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        for (int i=0; i<n; i++)
        {
            int required = target-nums[i];
            if (map.containsKey(required))
            {
                return new int[]{map.getOrDefault(required, 0), i};
            }
            map.put(nums[i], i);
        }
        return new int[0];
    }
}